ci/pipeline/ray_release_test.py

```diff
@@ -13,9 +20,45 @@ def main() -> int:
     logger = get_logger()
     changed_files = _get_changed_files()
     logger.info(f"Changed files: {changed_files}")
+    test_targets = _get_test_targets_for_changed_files(changed_files)
+    logger.info(test_targets)
     return 0
 
 
+def _get_test_targets_for_changed_files(changed_files: List[str]) -> str:
+    """
+    Get the test target for the changed files.
+    """
```

Comment:
so.. this is determining what tests to run from a past coverage data? when will this be used? will this also be used on branch/master postmerge or only on pr's?

how about tests that are newly added in a PR? will they get run or not?

Reply:
To bootstrap this new approach, I'm thinking starting with at the pr level. branch/master still runs full coverage as they are right now. As we gain trust in this new approach we can optimize further, but fixing pr level itself I think will already save a lot of costs.

There is only one case where this approach will fall short, which you exactly pointed out: newly added test. Newly added test will always get run using existing rules and not affected by coverage.